### PR TITLE
fix for Build.Repository.Name in multi-checkout scenario

### PR DESCRIPTION
**Pull request by jpricket**
Still need to test the scenarios.

### CODE REVIEW

File: src/Agent.Sdk/Util/RepositoryUtil.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@ namespace Microsoft.VisualStudio.Services.Agent.Util
     public static class RepositoryUtil
     {
         public static readonly string IsPrimaryRepository = "system.isprimaryrepository";
+        public static readonly string IsTriggeringRepository = "system.istriggeringrepository";
```
Comment (alex-peck) [inline src/Agent.Sdk/Util/RepositoryUtil.cs]: Does the server currently send this down? Or will it need to be added?

File: src/Agent.Sdk/Util/RepositoryUtil.cs [hunk 2]
```diff
@@ -50,10 +51,26 @@ public static bool IsPrimaryRepositoryName(string repoAlias)
...
+        /// If the list contains more than one, then we look for the repository marked as the primary repo.
         /// It returns null, if no primary repository can be found.
         /// </summary>
         public static RepositoryResource GetPrimaryRepository(IList<RepositoryResource> repositories)
+        {
+            return GetWellKnownRepository(repositories, RepositoryUtil.IsPrimaryRepository);
+        }
+
+        /// <summary>
+        /// This method returns the repo from the list that is considered the primary repository.
```
Comment (alex-peck) [inline src/Agent.Sdk/Util/RepositoryUtil.cs]: Comment copied from above? `triggering repository`

File: src/Agent.Sdk/Util/RepositoryUtil.cs (hunk 1 above)
Comment (jpricket) [reply src/Agent.Sdk/Util/RepositoryUtil.cs]: No, this flag is completely new. For now it should default to 'self' or the single repo. We can change that behavior in the future in a couple ways.
1. Figure out the triggering repo in the agent based on some information we already have and set the flag as early as possible.
2. Let the server set this flag on the repository when it sends us the job message.

In the case of number 2, we may want to revisit the name of the flag. I was assuming we could probably keep the flag internal to the agent and do number 1, but number 2 may actually be better.

File: src/Agent.Sdk/Util/RepositoryUtil.cs (hunk 2 above)
Comment (jpricket) [reply src/Agent.Sdk/Util/RepositoryUtil.cs]: thanks!!

File: src/Agent.Worker/Build/BuildJobExtension.cs [hunk 3]
```diff
@@ -143,17 +142,17 @@ public override void InitializeJobExtension(IExecutionContext executionContext,
                 throw new Exception(StringUtil.Loc("SupportedRepositoryEndpointNotFound"));
             }
 
-            executionContext.Debug($"Primary repository: {repoInfo.Repository.Properties.Get<string>(Pipelines.RepositoryPropertyNames.Name)}. repository type: {repoInfo.Repository.Type}");
+            executionContext.Debug($"Primary repository: {repoInfo.TriggeringRepository.Properties.Get<string>(Pipelines.RepositoryPropertyNames.Name)}. repository type: {repoInfo.TriggeringRepository.Type}");
```
Comment (alex-peck) [inline src/Agent.Worker/Build/BuildJobExtension.cs]: Do we want to change this debug text from "Primary repository" given that it now means something different?

Comment (jpricket) [reply src/Agent.Worker/Build/BuildJobExtension.cs]: absolutely, I will fix it

File: src/Agent.Sdk/Util/RepositoryUtil.cs [hunk 4]
```diff
@@ -50,10 +51,26 @@ public static bool IsPrimaryRepositoryName(string repoAlias)
...
+        /// If the list only contains 1 repo, then that is the primary repository.
+        /// If the list contains more than one, then we look for the repository marked as the primary repo.
+        /// It returns null, if no primary repository can be found.
+        /// </summary>
+        public static RepositoryResource GetTriggeringRepository(IList<RepositoryResource> repositories)
+        {
+            return GetWellKnownRepository(repositories, RepositoryUtil.IsTriggeringRepository);
+        }
+
+        private static RepositoryResource GetWellKnownRepository(IList<RepositoryResource> repositories, string repositoryFlagName)
```
Comment (damccorm) [inline src/Agent.Sdk/Util/RepositoryUtil.cs]: I like this more generalized approach!

**Pull request merged**